### PR TITLE
chore(auth): Consolidate Android/iOS code

### PR DESCRIPTION
**Pull request by dnys1**
- Removes platform packages
- Consolidates macOS/iOS code in `darwin` directory
- Migrate to Swift/Kotlin pigeons

### CODE REVIEW

File: packages/auth/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/NativeAuthPluginBindingsPigeon.kt
Comment (Jordan-Nelson) [inline packages/auth/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/NativeAuthPluginBindingsPigeon.kt:1]: Do you want to either put this in a /pigeons dir or add *Pigeon.kt to .gitattributes as generated code?

File: packages/auth/amplify_auth_cognito/darwin/Classes/messages.g.swift
Comment (Jordan-Nelson) [inline packages/auth/amplify_auth_cognito/darwin/Classes/messages.g.swift:1]: pigeons dir or update .gitattributes. Aligning on one pattern would be nice as well. I think we could use .g.kt as well.

**Review by Jordan-Nelson — APPROVED** [review]

File: packages/auth/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/NativeAuthPluginBindingsPigeon.kt
Comment (dnys1) [reply packages/auth/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/NativeAuthPluginBindingsPigeon.kt:1]: Lol whoops. I didn't mean to introduce yet another naming convention 😆

File: packages/auth/amplify_auth_cognito/example/ios/Podfile [hunk 1]
```diff
@@ -36,14 +36,14 @@ target 'Runner' do
...
-    inherit! :complete
-  end
 end
 
 post_install do |installer|
   installer.pods_project.targets.each do |target|
     flutter_additional_ios_build_settings(target)
+    # Workaround for: https://github.com/flutter/flutter/issues/126449
+    target.build_configurations.each do |config|
+      config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '13.0'
```
Comment (Equartey) [inline packages/auth/amplify_auth_cognito/example/ios/Podfile:46]: This is no longer needed, right?

**Review by Jordan-Nelson — APPROVED** [review]

File: packages/auth/amplify_auth_cognito/example/ios/Podfile (hunk 1 above)
Comment (dnys1) [reply packages/auth/amplify_auth_cognito/example/ios/Podfile:46]: Correct. We can remove for all packages in a separate PR.

**Review by Equartey — COMMENTED** [review]
Thanks for cleaning up! Just one comment

**Review by Equartey — APPROVED** [review]

**Pull request merged**